lib/evaluation/filterShouldExistIfPredicateExistsEvaluationHandler.ts

```diff
@@ -0,0 +1,43 @@
+import { Field, InputDescriptor, PresentationDefinition } from '@sphereon/pe-models';
+
+import { Status } from '../ConstraintUtils';
+
+import { AbstractEvaluationHandler } from './abstractEvaluationHandler';
+import { HandlerCheckResult } from './handlerCheckResult';
+
+export class FilterShouldExistIfPredicateExistsEvaluationHandler extends AbstractEvaluationHandler {
+  public getName(): string {
+    return 'FilterShouldExistIfPredicateExists';
+  }
+
+  public handle(pd: PresentationDefinition, _vp: any, results: HandlerCheckResult[]): void {
+    for (let i = 0; i < pd.input_descriptors.length; i++) {
+      const inputDescriptor: InputDescriptor = pd.input_descriptors[i];
+      if (inputDescriptor.constraints && inputDescriptor.constraints.fields) {
+        for (let j = 0; j < inputDescriptor.constraints.fields.length; j++) {
```

Comment:
- If you won't change the index, make it a constant.
- You can use for (const [index, field] of inputDescriptor.constraints.fields.entries()). It's less verbose.

Reply:
deleted the handler